src/java/picard/reference/NonNFastaSize.java

```diff
@@ -0,0 +1,103 @@
+package picard.reference;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.samtools.reference.ReferenceSequence;
+import htsjdk.samtools.reference.ReferenceSequenceFile;
+import htsjdk.samtools.reference.ReferenceSequenceFileFactory;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.samtools.util.IntervalList;
+import htsjdk.samtools.util.IntervalListReferenceSequenceMask;
+import htsjdk.samtools.util.ReferenceSequenceMask;
+import htsjdk.samtools.util.StringUtil;
+import htsjdk.samtools.util.WholeGenomeReferenceSequenceMask;
+import picard.PicardException;
+import picard.cmdline.CommandLineProgram;
+import picard.cmdline.CommandLineProgramProperties;
+import picard.cmdline.Option;
+import picard.cmdline.StandardOptionDefinitions;
+import picard.cmdline.programgroups.Fasta;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * A tool to count the number of non-N bases in a fasta file
+ */
+@CommandLineProgramProperties(
+        usage = NonNFastaSize.USAGE_SUMMARY + NonNFastaSize.USAGE_DETAILS,
+        usageShort = NonNFastaSize.USAGE_SUMMARY,
+        programGroup = Fasta.class
+)
+public class NonNFastaSize extends CommandLineProgram {
+    static final String USAGE_SUMMARY = "Counts the number of non-N bases in a fasta file.";
+    static final String USAGE_DETAILS = "This tool takes any FASTA-formatted file and counts the number of non-N bases in it." +
+            "Note that it requires that the fasta file have associated index (.fai) and dictionary (.dict) files.<br />" +
+            "<h4>Usage example:</h4>" +
+            "<pre>" +
+            "java -jar picard.jar NonNFastaSize \\<br />" +
+            "      I=input_sequence.fasta \\<br />" +
+            "      O=count.txt" +
+            "</pre>" +
+            "<hr />"
+            ;
+    @Option(shortName= StandardOptionDefinitions.INPUT_SHORT_NAME, doc="The input FASTA file.")
+    public File INPUT;
+
+    @Option(shortName= StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc="The output file in which to record the count.")
+    public File OUTPUT;
+
+    @Option(shortName = "INTERVALS", doc = "An interval list file that contains the locations of the positions to assess.  If not provided, the entire reference will be used", optional = true)
+    public File INTERVALS = null;
```

Comment:
I got dinged in a previous PR (not by you) for not having the null called out explicitly!
Would prefer to keep it here therefore.